### PR TITLE
Addon-docs: Use `.mdx` globs in templates and defaults

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: N/A

## What I did

- Update CLI templates to use `.mdx` files and an `.mdx` globs (no more `.stories.mdx`).
- Updated globs used by default in stories specifiers to be a little simpler. Still match both `.mdx` and `.stories.mdx`. I think we might consider matching just `.mdx`.

## TODO

Decide on the above. Also add migration note either way (we are going to start matching new `.mdx` files, and maybe stop matching `.stories.mdx` files).

## How to test

Generate a new template and check it includes an introduction and works.

### CODE REVIEW

File: code/lib/core-common/src/utils/__tests__/normalize-stories.test.ts [hunk 1]
```diff
@@ -230,7 +230,7 @@ describe('normalizeStoriesEntry', () => {
       {
         "titlePrefix": "",
         "directory": ".",
-        "files": "**/*.@(mdx|stories.mdx|stories.tsx|stories.ts|stories.jsx|stories.js)",
+        "files": "**/*.@(mdx|stories.@(tsx|ts|jsx|js))",
```
Comment (shilman) [inline code/lib/core-common/src/utils/__tests__/normalize-stories.test.ts:233]: Whoa. Is this a valid glob?

Comment (tmeasday) [reply code/lib/core-common/src/utils/__tests__/normalize-stories.test.ts:233]: It seems to work! So I guess so.

Comment (tmeasday) [reply code/lib/core-common/src/utils/__tests__/normalize-stories.test.ts:233]: We've been using it for a while in our sandboxes: https://github.com/storybookjs/storybook/blob/7f9e995af04c11e570cc5f5e18737491fff60dcb/scripts/tasks/sandbox-parts.ts#L258

**Comment by tmeasday** [comment]
@shilman I changed my mind, maybe we should do a migration here to replace
 a) `stories: ['./directory']` or 
 b) `stories: [{ directory: 'xyz' }]`

with
 a) `stories: [{ directory: 'xyz', files: '**/*.stories.@(mdx|tsx|ts|jsx|js)' }]`

Then for we can agressively change the default ("unspecified") regexp to not include `.stories.mdx` files?

**Comment by shilman** [comment]
@tmeasday I like that more aggressive approach. It would also give us a chance to explain things very clearly in the migration step

**Comment by tmeasday** [comment]
@shilman the only problem with the aggressive approach is that we would very much need to gate the migration on the 6.x->7.x migration -- otherwise it will run again incorrectly for future 7.x users. 

I'll create a ticket about that and put it in our "automigrations" project -- if we build said versioned mechanism before 7.0 we can do that migration, otherwise we'll have to stick with the simpler approach of this PR (unless you have a better idea).

**Review by shilman — APPROVED** [review]
LGTM! We can potentially pursue the more aggressive route later

**Pull request merged**